File: src/plugins/core/scripting/files.cpp
#include "../scripting.h"
#include "../../../files/Files.h"

PluginFiles::PluginFiles(std::string m_plugin_name)
{
    this->plugin_name = m_plugin_name;
}

void PluginFiles::Append(std::string path, std::string content, bool hasdate)
{
    Files::Append(path, content, hasdate);
}

void PluginFiles::Write(std::string path, std::string content, bool hasdate)
{
    Files::Write(path, content, hasdate);
}

std::string PluginFiles::Read(std::string path)
{
    return Files::Read(path);
}

void PluginFiles::Delete(std::string path)
{
    Files::Delete(path);
}

std::string PluginFiles::GetBase(std::string filePath)
{
    return Files::getBase(filePath);
}

bool PluginFiles::ExistsPath(std::string path)
{
    return Files::ExistsPath(path);
}

bool PluginFiles::IsDirectory(std::string path)
{
    return Files::IsDirectory(path);
}

std::vector<std::string> PluginFiles::FetchFileNames(std::string path)
{
    return Files::FetchFileNames(path);
}

std::vector<std::string> PluginFiles::FetchDirectories(std::string path)
{
    return Files::FetchDirectories(path);
}

bool PluginFiles::CreateDirectory(std::string path)
{
    return Files::CreateDirectory(path);
}

bool PluginFiles::Compress(std::string path, std::string output)
{
    return Files::Compress(path, output);
}
bool PluginFiles::Decompress(std::string path, std::string output)
{
    return Files::Decompress(path, output);
}

File: src/managed/API/Scripting/Files.cs
using SwiftlyS2.Internal_API;

namespace SwiftlyS2.API.Scripting
{
    public class Files
    {
        private static IntPtr _ctx = IntPtr.Zero;
    
        private static void InitializeContext()
        {
            if (_ctx != IntPtr.Zero) return;
            _ctx = Invoker.CallNative<IntPtr>("Files", "Files", CallKind.ClassFunction);
        }
        public static void Append(string path, string content, bool hasdate)
        {
            InitializeContext();
            Internal_API.Invoker.CallNative("Files", "Append", Internal_API.CallKind.ClassFunction, _ctx, path, content, hasdate);
        }
        public static bool Compress(string path, string output)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<bool>("Files", "Compress", Internal_API.CallKind.ClassFunction, _ctx, path, output);
        }
        public static bool CreateDirectory(string path)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<bool>("Files", "CreateDirectory", Internal_API.CallKind.ClassFunction, _ctx, path);
        }
        public static bool Decompress(string path, string output)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<bool>("Files", "Decompress", Internal_API.CallKind.ClassFunction, _ctx, path, output);
        }
        public static void Delete(string path)
        {
            InitializeContext();
            Internal_API.Invoker.CallNative("Files", "Delete", Internal_API.CallKind.ClassFunction, _ctx, path);
        }
        public static bool ExistsPath(string path)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<bool>("Files", "ExistsPath", Internal_API.CallKind.ClassFunction, _ctx, path);
        }
        public static string[] FetchDirectories(string path)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<string[]>("Files", "FetchDirectories", Internal_API.CallKind.ClassFunction, _ctx, path);
        }
        public static string[] FetchFileNames(string path)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<string[]>("Files", "FetchFileNames", Internal_API.CallKind.ClassFunction, _ctx, path);
        }
        public static string GetBase(string path)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<string>("Files", "GetBase", Internal_API.CallKind.ClassFunction, _ctx, path);
        }
        public static bool IsDirectory(string path)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<bool>("Files", "IsDirectory", Internal_API.CallKind.ClassFunction, _ctx, path);
        }
        public static string Read(string path)
        {
            InitializeContext();
            return Internal_API.Invoker.CallNative<string>("Files", "Read", Internal_API.CallKind.ClassFunction, _ctx, path);
        }
        public static void Write(string path, string content, bool hasdate)
        {
            InitializeContext();
            Internal_API.Invoker.CallNative("Files", "Write", Internal_API.CallKind.ClassFunction, _ctx, path, content, hasdate);
        }
    }
}